crates/iceberg/src/spec/values.rs

```diff
@@ -449,7 +456,30 @@ impl Datum {
             PrimitiveLiteral::String(val) => ByteBuf::from(val.as_bytes()),
             PrimitiveLiteral::UInt128(val) => ByteBuf::from(val.to_be_bytes()),
             PrimitiveLiteral::Binary(val) => ByteBuf::from(val.as_slice()),
-            PrimitiveLiteral::Int128(_) => todo!(),
+            PrimitiveLiteral::Int128(val) => {
+                let PrimitiveType::Decimal { precision, .. } = self.r#type else {
+                    unreachable!(
```

Comment:
I think you're right  @liurenjie1024 . I don't see a way that a malformed `Datum` could be created anywhere right now but this may not always be true as the project continues to evolve.

Reply:
So we will change `Datum::to_bytes`'s return value into `Result<ByteBuf>`?